src/components/GameItem.js

```diff
@@ -4,6 +4,30 @@ import ItemIcons from '../ItemIcons.js';
 import PropTypes from 'prop-types';
 
 class GameItem extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      spotted: false,
+    };
+  }
+
+  onItemClick = () => {
+    this.setState({spotted: true});
+    if (this.props.type === 'litter') {
```

Comment:
One issue, I can run up my score by mashing the mouse on a single piece of litter.  You should also only call the callback, if `this.state.spotted` is false.